### DISCUSSION TITLE
Error in Upgrading from 3.7.0 to 3.7.1

### DISCUSSION THREAD

**Discussion by GSzarkaTAT** [Nitrux/nitrux-bug-tracker#193] [Q&A] [answered]
https://nxos.org/tutorial/how-to-upgrade-nitrux/

```
sudo nuts upgrade                                                                                                                     at 06:21:29 
[sudo] gabor jelszava: 
Nitrux Update Tool System: Info: Nitrux Update Tool System version b2e4e9d9347faf881e5995e8f67294da.
Nitrux Update Tool System: Success: Nitrux Update Tool System is running as root, continuing...
Nitrux Update Tool System: Success: Success! Nitrux Update Tool System configuration file found, continuing...
Nitrux Update Tool System: Info: Nitrux Update Tool System log file found, skipping...
~ ❯ sudo nuts update                                                                                                                      at 06:21:38 
Nitrux Update Tool System: Info: Nitrux Update Tool System version b2e4e9d9347faf881e5995e8f67294da.
Nitrux Update Tool System: Success: Nitrux Update Tool System is running as root, continuing...
Nitrux Update Tool System: Success: Success! Nitrux Update Tool System configuration file found, continuing...
Nitrux Update Tool System: Info: Nitrux Update Tool System log file found, skipping...
Nitrux Update Tool System: Success: This utility is running in Nitrux, continuing...
Nitrux Update Tool System: Success: This computer can reach the Internet, continuing...
Nitrux Update Tool System: Info: Can't find nuts-query.info, downloading...
Nitrux Update Tool System: Info: An update is available for Nitrux 3.7.0, continuing...
Nitrux Update Tool System: Info: All required directories exist, continuing...
Nitrux Update Tool System: Info: Redundant backup of NX_ROOT found, skipping...
Nitrux Update Tool System: Info: Found overlayroot, continuing...
ERROR: Unable to find an overlayroot filesystem
```
Operating System: Nitrux 3.7.0 build.300924.ub
KDE Plasma Version: 5.27.11
KDE Frameworks Version: 5.115.0
Qt Version: 5.15.13
Kernel Version: 6.10.12-1-liquorix-amd64 (64-bit)
Graphics Platform: Wayland
Processors: 4 × Intel® N100
Memory: 7.5 GiB of RAM
Graphics Processor: Mesa Intel® Graphics
Manufacturer: GMKtec
Product Name: NucBox G3

**Comment by UriHerrera** [comment]
As mentioned in the tutorial, the error message occurs when the overlay is unavailable. 

What is the output of `mount | grep overlay`? If it doesn't match the output below, the overlay is disabled.

```
overlayroot on / type overlay (rw,relatime,lowerdir=/media/root-ro,upperdir=/media/root-rw/overlay,workdir=/media/root-rw/overlay-workdir/_,uuid=on)
overlayroot on /System/Boot type overlay (rw,relatime,lowerdir=/media/root-ro,upperdir=/media/root-rw/overlay,workdir=/media/root-rw/overlay-workdir/_,uuid=on)
overlayroot on /System/Configuration type overlay (rw,relatime,lowerdir=/media/root-ro,upperdir=/media/root-rw/overlay,workdir=/media/root-rw/overlay-workdir/_,uuid=on)
overlayroot on /System/Binaries/Administrative type overlay (rw,relatime,lowerdir=/media/root-ro,upperdir=/media/root-rw/overlay,workdir=/media/root-rw/overlay-workdir/_,uuid=on)
overlayroot on /System/Binaries/Standard type overlay (rw,relatime,lowerdir=/media/root-ro,upperdir=/media/root-rw/overlay,workdir=/media/root-rw/overlay-workdir/_,uuid=on)
overlayroot on /System/Binaries/Optional type overlay (rw,relatime,lowerdir=/media/root-ro,upperdir=/media/root-rw/overlay,workdir=/media/root-rw/overlay-workdir/_,uuid=on)
overlayroot on /System/Filesystems/Internal type overlay (rw,relatime,lowerdir=/media/root-ro,upperdir=/media/root-rw/overlay,workdir=/media/root-rw/overlay-workdir/_,uuid=on)
overlayroot on /System/Filesystems/External type overlay (rw,relatime,lowerdir=/media/root-ro,upperdir=/media/root-rw/overlay,workdir=/media/root-rw/overlay-workdir/_,uuid=on)
overlayroot on /System/Libraries type overlay (rw,relatime,lowerdir=/media/root-ro,upperdir=/media/root-rw/overlay,workdir=/media/root-rw/overlay-workdir/_,uuid=on)
overlayroot on /System/Shareable type overlay (rw,relatime,lowerdir=/media/root-ro,upperdir=/media/root-rw/overlay,workdir=/media/root-rw/overlay-workdir/_,uuid=on)
overlayroot on /System/Variable type overlay (rw,relatime,lowerdir=/media/root-ro,upperdir=/media/root-rw/overlay,workdir=/media/root-rw/overlay-workdir/_,uuid=on)
```

Also, include the content of `/etc/fstab`; run the command `cat /etc/fstab` and the output of `cat /proc/cmdline` as your user; don't use `sudo`

**Comment by GSzarkaTAT** [comment]
Hi Uri,
Thanks for your help.
Below is the output you requested from Station. Honestly, I don't know what the "overlay" means, and how to switch it on...maybe you can give me some hints how to switch it on? (a link where it is written)

```
~ ❯ mount | grep overlay                                                                                                                  at 05:05:51 
~ ❯ sudo mount | grep overlay                                                                                                       ✘ 0|1 at 05:06:12 
[sudo] gabor jelszava: 
~ ❯ cat /etc/fstab                                                                                                                  ✘ 0|1 at 05:06:36 
# /etc/fstab: static file system information.
#
# Use 'blkid' to print the universally unique identifier for a device; this may
# be used with UUID= as a more robust way to name devices that works even if
# disks are added and removed. See fstab(5).
#
# <file system>             <mount point>  <type>  <options>  <dump>  <pass>
UUID=30b0c60f-dd0e-4825-978f-a96227e1ff72 /              ext4    defaults,discard 0 1
tmpfs                                     /tmp           tmpfs   defaults,noatime,mode=1777 0 0
~ ❯ cat /proc/cmdline                                                                                                                     at 05:07:50 
audit=0 intel_pstate=disable  BOOT_IMAGE=/boot/vmlinuz-6.10.12-1-liquorix-amd64 root=UUID=30b0c60f-dd0e-4825-978f-a96227e1ff72 quiet splash
~ ❯                                                                                                                                       at 05:08:05 
```

**Comment by UriHerrera** [comment]
> Hi Uri,
> Thanks for your help.
> Below is the output you requested from Station. Honestly, I don't know what the "overlay" means, and how to switch it on...maybe you can give me some hints how to switch it on? (a link where it is written)

You don't have to do anything. When the distribution is installed as intended, all this is set up by default.

---

The cutoff output from your `/etc/fstab` shows the default state of the file. 

> UUID=30b0c60f-dd0e-4825-978f-a96227e1ff72 / ext4 defaults,discard 0 1

Unfortunately, this line indicates that you didn't follow the installation tutorial, which is available offline in the Documentation folder and online in the Tutorials section on the website. 

We do not use the EXT4 filesystem when setting up partitions or mention using it in our documentation.

I don't know if you did manual partitioning or selected the option "Install alongside" (which we specifically mentioned not to use).

Could you tell me how you specifically installed the distribution?

---

> ~ ❯ cat /proc/cmdline                                                 at 05:07:50
>
> audit=0 intel_pstate=disable BOOT_IMAGE=/boot/vmlinuz-6.10.12-1-liquorix-amd64 root=UUID=30b0c60f-dd0e-4825-978f-a96227e1ff72 quiet splash

It's also worth mentioning that your output of `cat /proc/cmdline` does not match our default parameters. 

So, yes, you did not install the distribution correctly. The solution to this problem is to follow the installation tutorial precisely as it is.

Please read __all of our documentation__ thoroughly.

**Comment by GSzarkaTAT** [comment]
Hi Uri,
My Nitrux is the 3rd operating system after first installing Windows 11 and then Ubuntu. I have used manual partitioning, so I setup exactly how CoPilot and ChatGPT told me. I have struggled a lot to get GRUB and to be able to select from 3 of them because of different systems. :-(
All 3 systems are working correctly on a single disk. I have no experience on Linux however I admit. I am just curious about new operating system and tried Nitrux (but not via Live CD since the changes are lost after restart). I am not planning to have only Nitrux on my mini PC.
Looking at the Documentation "Dual-booting with Windows" - yes, I had to disable Secure boot (I don't know what benefit I have under Windows, but having it enabled, Ventoy could not be used for installing Ubuntu/Nitrux), and yes, I shrinked existing partition to install Nitrux via Live CD. That's it. After this I just had to fix the GRUB bootloder. Of course, I have downloaded the ISO for Nitrux from your website (both 3.7.0 and 3.7.1). 
However, Looking at the section "Installing Nitrux", I am not sure I understand how to fix "EXT4" partition issue, if that is really a blocking point. 
Is there any way for me to upgrade to 3.7.1 or should I wait for maybe 4.0 ? Tell me. Thanks.

**Comment by UriHerrera** [comment, marked as answer]
I'm moving this to a discussion since it's not a bug.

---

> However, Looking at the section "Installing Nitrux", I am not sure I understand how to fix "EXT4" partition issue, if that is really a blocking point.

It is indeed a blocking point because we don't use EXT4. You only need to follow the tutorial's instructions: delete the partition you've created and create the partitions outlined in it, or select "Replace partition."

The bootloader of the last system you install will take over the bootloading duties. Installing Nitrux last will add the entries for Windows and Ubuntu. However, some distributions don't use GRUB. 

Anyway, you can always boot whichever OS you want using the computer's EFI boot menu.

**Comment by GSzarkaTAT** [comment]
Finally, I gave a try to install 3.7.1 from scratch. I downloaded ISO and started installation. I selected Manual partition.  Before this I recognized that Replacing currect partion suggested  to establish 3 sections  like NX_ROOT, NX_HOME and NX_VAR_LIB. I followed the suggestion.
When installation was successfully completed I restarted PC.  It did not boot. It misses boot kernel. 
![20241108_192032](https://github.com/user-attachments/assets/3af61cc1-ad43-4ffa-ad15-a8ef063723ea)

Remark: maybe I forgot to update grub???
![20241108_192201](https://github.com/user-attachments/assets/a417bc78-9c57-446e-aca5-83163936e8e9)


At this point I gave up. 
I now installed another distro without problems. I just had to update grub.

**Reply by UriHerrera** [reply]
> Remark: maybe I forgot to update grub???

Without any logs, I have no idea. If I install the distribution, it will work fine.

> ![imagen](https://github.com/user-attachments/assets/61f11453-1602-4105-be2a-830730011a21)

Nitrux 3.7.1 doesn't use Linux 6.10.12. In your screenshot, the problem is that the bootloader tried to boot a non-existent partition. I'm going to assume it's the partition you've deleted. Also, I don't know whose GRUB that is since you've mentioned you also had Ubuntu. Still, it's certainly not ours, given that the distribution was installed correctly in your other screenshot, and it wouldn't be pointing to a deleted partition.
